### PR TITLE
Enable using DHCP client and server in LWIP

### PR DESCRIPTION
**Pull request by SeonghoByeon**
1. commit 37f123eadf85580d8a1fbb755bec3d972973e331
Modify dhcp client and server function calls, using the newly implemented one
    - artik_demo
    - itc_dm_main
    - wakama_client
    - netcmd (dhcpc, dhcpd)
    - slsi_dm_connectivity

2. commit f740afe79cb0bf74fa51d89e9cbf9e7b115f85b1
Add CONFIG_NET_DHCP definition in defconfigs
   - Add CONFIG_NET_DHCP=y if CONFIG_NETUTILS_DHCPC or DHCPD is used

3. commit c76011c513be5dd3042a4e0c3ab86f2628108419
framework/wifi_manager: Modify DHCPC and DHCPS call logic
    - Use the newly released netutils dhcp for both client and server

4. commit 5079c19662d432ed8142d8d48a9129e1b7c10e14
external/dhcp: Implement DHCP of LWIP
    - Implement functions that utilize DHCP of LWIP, depending on configuration
    - Add netifapis used for DHCP

5. commit b8a7eeb0a55db16f1d506e5841e0de20591a84fb
os/net/lwip: Fix bug for IP addr management and add user callbak in DHCPS
    - Modify DHCPS to differentiate in_addr_t structure depending on ipv6
    - Add a global variable pointing out staion join callback in dhcps
    - Define user callback and make registering function in dhcps

6. commit 56a7fd8578096e1251adb9cfe0a2475e3155135a
os/net/lwip: Arrange DHCP and DHCPS config
    - Add NET_DHCP to select external DHCP app or NET_LWIP_DHCP to select internal LWIP DHCP
    - Arrange Kconfig related with DHCP to clearly represent in Networking Support / Protocol

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [37f123eadf85580d8a1fbb755bec3d972973e331] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [37f123eadf85580d8a1fbb755bec3d972973e331] - Code Rule Check OK.

File: external/dhcpc/dhcpc_lwip.c [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * netutils/dhcpc/dhcpc_lwip.c
```
Comment (thapav) [inline external/dhcpc/dhcpc_lwip.c]: Please correct file path

File: external/dhcpd/dhcpd_lwip.c [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * netutils/dhcpd/dhcpd.c
```
Comment (thapav) [inline external/dhcpd/dhcpd_lwip.c]: Please correct file path

File: external/dhcpd/dhcpd_lwip.c [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
```
Comment (thapav) [inline external/dhcpd/dhcpd_lwip.c]: change year to 2018

**Pull request merged**